### ISSUE TITLE
Parser error on RAP syntax with `%` in identifiers

### ISSUE THREAD

**Issue by agoeb** [abaplint/abaplint#3177]
The following snippet leads to a parser error in abapLint

```
LOOP AT lt_my_table REFERENCE INTO DATA(ls_item).
  APPEND VALUE #( %tky        = ls_item->%tky
                  %action-foo = COND #( WHEN ls_item->status = 1
                                        THEN if_abap_behv=>fc-o-enabled
                                        ELSE if_abap_behv=>fc-o-disabled ) ) TO rt_result.
ENDLOOP.
```

> Statement does not exist in ABAPv757(or a parser error), "APPEND"(parser_error)

This in turn leads to a false-positive finding for the `empty_structure` check on the `LOOP`.

It seems the issue is the RAP-style `%tky` and `%action` identifiers. If I remove the `%`s in the abaplint playground, the parser error is gone, and the `empty_structure` finding disappears as well.
